### PR TITLE
Made a base class for ZeroPadding.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
### Summary

By doing this, we can remove quite a lot of code. I think it will also increase the coverage. (maybe?)

A similar refactoring was done for the cropping layers.

### PR Overview

The base class holds the `compute_output_shape` and `get_config` methods.

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/layers/convolutional.py [hunk 1]
```diff
@@ -2040,7 +2040,50 @@ def get_config(self):
         return dict(list(base_config.items()) + list(config.items()))
 
 
-class ZeroPadding1D(Layer):
+class _ZeroPadding(Layer):
```
Comment (fchollet) [inline keras/layers/convolutional.py]: If this base layer does not have a `call` method, it should implement `call` and raise `NotImplementedError` there.

File: keras/layers/convolutional.py [hunk 2]
```diff
@@ -2060,29 +2103,22 @@ class ZeroPadding1D(Layer):
     """
 
     def __init__(self, padding=1, **kwargs):
```
Comment (fchollet) [inline keras/layers/convolutional.py]: Note that we could consider adding a `data_format` argument for the 1D version as well (in a later PR).

File: keras/layers/convolutional.py [hunk 3]
```diff
@@ -2040,7 +2040,53 @@ def get_config(self):
...
+    def __init__(self, padding, data_format=None, **kwargs):
+        # self.rank is 1 for ZeroPadding1D, 2 for ZeroPadding2D.
+        self.rank = len(padding)
+        self.padding = padding
+        self.data_format = K.normalize_data_format(data_format)
+        self.input_spec = InputSpec(ndim=self.rank + 2)
+        super(_ZeroPadding, self).__init__(**kwargs)
+
+    def call(self, inputs):
+        return NotImplementedError
```
Comment (joelthchao) [inline keras/layers/convolutional.py]: `raise NotImplementedError`?

Comment (gabrieldemarmiesse) [reply keras/layers/convolutional.py]: Indeed.

**Comment by gabrieldemarmiesse** [comment]
`test_regression_class_build_fn` is giving no output for more than 10 minutes on Theano, causing the build to fail. See #10995 .

**Comment by fchollet** [comment]
Restarting tests.

**Comment by gabrieldemarmiesse** [comment]
It seams unrelated tests are failing.

**Comment by fchollet** [comment]
It seems we have a recurring timeout issue on Travis (cc @taehoonlee). I guess we could try to diff the passing tests [here](https://travis-ci.org/keras-team/keras/jobs/421671864) with the passing tests on a successful run, and try to figure out which tests could be causing this.

**Comment by taehoonlee** [comment]
@fchollet, Recently, as you may know, the timeout has been arising even in CNTK and Theano. We are trying, but it is not so easy to investigate the timeout issue. The pattern is so random :( 

We have no choice but just to assume that 9 jobs share computational resources and the last job left suffers from the timeout issue due to the lack of memory. Don't we?

**Pull request merged**